qa/evil-tests/src/test/java/org/elasticsearch/plugins/InstallPluginCommandTests.java

```diff
@@ -457,6 +460,29 @@ public void testConfigConflict() throws Exception {
         assertInstallCleaned(env);
     }
 
+    public void testMissingDescriptor() throws Exception {
+        Environment env = createEnv();
+        Path pluginDir = createTempDir();
+        Files.createFile(pluginDir.resolve("fake.yml"));
+        String pluginZip = writeZip(pluginDir, PathUtils.get("elasticsearch"));
+        expectThrows(NoSuchFileException.class, () -> {
+            installPlugin(pluginZip, env);
+        });
```

Comment:
Thanks for adding this test. We should check the exception is what we expect, that the message contains plugin-descriptor.properties